src/screens/EventsScreen/ResetAllFiltersButton.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+import React from "react";
+import { Animated, StyleSheet, Easing } from "react-native";
+import type { ViewLayoutEvent } from "react-native/Libraries/Components/View/ViewPropTypes";
+import FilterHeaderButton from "./FilterHeaderButton";
+import text from "../../constants/text";
+
+type Props = {
+  visible: boolean,
+  onPress: () => void,
+  animationTime: number,
+  animationDelay: number
+};
+
+type State = {
+  isAnimating: boolean
+};
+
+const DEFAULT_HEIGHT = 120;
+const DEFAULT_FADE_VALUE = 1;
+const DEFAULT_TOP_OFFSET_VALUE = 0;
+
+class ResetAllFiltersButton extends React.Component<Props, State> {
+  static defaultProps = {
+    animationTime: 200,
+    animationDelay: 50
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      isAnimating: false
+    };
+  }
+
+  setButtonHeight = (e: ViewLayoutEvent): void => {
+    const { height } = e.nativeEvent.layout;
+    this.height = height;
+  };
+
+  fadeOut = (): void => {
+    const { animationTime, animationDelay } = this.props;
+
+    Animated.parallel([
+      Animated.timing(this.fadeValue, {
+        toValue: 0,
+        duration: animationTime / 2
+      }),
+      Animated.timing(this.topOffset, {
+        toValue: -this.height,
+        duration: animationTime,
+        easing: Easing.out(Easing.quad),
+        delay: animationDelay
+      })
+    ]).start(this.animationFinished);
+  };
+
+  resetAllFilters = (): void => {
+    this.props.onPress();
+    this.setState({ isAnimating: true });
+    this.fadeOut();
+  };
+
+  resetAnimation = (): void => {
+    this.topOffset = new Animated.Value(DEFAULT_TOP_OFFSET_VALUE);
+    this.fadeValue = new Animated.Value(DEFAULT_FADE_VALUE);
+  };
+
+  animationFinished = (): void => {
+    this.setState({ isAnimating: false });
```

Comment:
This might be called when the button has already been unmounted, in which case `setState` will throw an exception.

Reply:
Ah yeah good catch.  I notice that isMounted() is now depricated. Do you have a pattern in mind to catch this? Maybe a try catch (might be a bit crude).

Reply:
Looked into this a bit more and there is a post from Dan Abramov on this issue:

https://github.com/facebook/react/issues/2787#issuecomment-304485634

```
In any case, React doesn’t throw errors when this happens anymore. But we will intentionally keep the suboptimal case clunky so that the developers have incentives to implement the right solution (cancellation).
```

It is now only a warning and not an error so inclined to leave this as is due do it being a edgy edge case unless you think it will be a major issue?

Reply:
Hm, interesting. We tested this we @RGBboy for the heart animation. I'm pretty sure we could make this crash. 🤔 

As far as I can see the recommendation is to set a field, which is basically what we ended up doing here:

https://github.com/redbadger/pride-london-app/blob/3a94242f84d188f81c9f3f1db238679069d094f0/src/components/SaveEventButton.js#L52-L54

Maybe our crash was unrelated though and we can just leave this at it is.

Reply:
I created a unit test for `unmount` -> `setState` and it threw an exception. So I just added a quick fix for this, just to be on the safe side.